src/libs/antares/study/parts/thermal/cluster.h

```diff
@@ -349,6 +381,18 @@ class ThermalCluster final : public Cluster, public std::enable_shared_from_this
     */
     double* productionCost;
 
+    /*!
+    ** \brief Production Cost, Market Bid Cost and Marginal Cost Matrixes - Per Hour and per Time
+    *Series
+    */
+    struct ThermalEconomicTimeSeries
+    {
+        std::array<double, HOURS_PER_YEAR> productionCostTs;
```

Comment:
Add `fuelCost` & `co2cost` here instead of `prepro`

Reply:
productionCostTs, marketBidCostPerHourTs & marginalCostPerHourTs all have same number of TS.
On the other hand fuelCost & co2cost can have different number of TS.
More importantly fuelCost & co2cost must be of type Matrix<double> cause functions like:
fuelcost.loadFromCSVFile
fuelcost.saveToCSVFile
fuelcost.markAsModified()
...
are used in this solution for reading/writing these matrixes into txt files.

Reply:
I like this solution, please include it.

Reply:
Yes, keep Matrix types for `fuelCost` and `co2cost`, just move them out of `prepro`